src/edit.js

```diff
@@ -90,9 +115,16 @@ const userDropdown = () => {
 		<>
 			<InspectorControls>
                 <PanelBody title="Users">
-                    <DropdownMenu
+                    <SelectControl
                         label="Users"
-                        controls={ users.map(user => ( {title: user.name, onClick: userDropdown } ))}
+                        options={ mappedUsers }
+                        value={ selectedUser?.id || 0 }
+                        onChange={ ( userId ) => {
+                            const userFromUsers = mappedUsers.find( ( user ) => {
+                                return user.id === parseInt( userId );
+                            } );
+                            setSelectedUser( userFromUsers );
```

Comment:
This will set the list for displaying in the `edit` view since we are using `useState`.

If you want to set a value for `userFromUsers` (terrible name I came up with 🤣 — something like `featuredUser` would be way better), you could create that property in `block.json` and then update it with `setAttribute` here instead of `setSelectedUser`.